daniel/pass/ps.c

```diff
@@ -0,0 +1,83 @@
+#include "ps.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <dirent.h>
+
+int_array getProcesses() {
+    DIR *directory;
+    int_array pidList;
+    int counter = 0;
+    struct dirent *file;
+    directory = opendir("/proc");
+    if (directory) {
+        while ((file = readdir(directory)) != NULL) {
```

Comment:
yikes
inline assignments are not really readable, maybe separate the file variable assignment from the condition